src/javascript/app_2/Stores/Modules/Contract/Helpers/__tests__/logic.js

```diff
@@ -0,0 +1,362 @@
+
+    describe('getEndSpotTime', () => {
+        it('should return contract\'s sell spot time if it is path dependent', () => {
+            const contract_info = {
+                "is_path_dependent": true,
+                "sell_spot_time": 123456,
+                "exit_tick_time": 987654321,
+            };
+            expect(Logic.getEndSpotTime(contract_info)).to.eql(123456);
+        });
+        it('should return contract\'s exit tick time if it is not path dependent', () => {
+            const contract_info = {
+                "is_path_dependent": false,
+                "sell_spot_time": 123456,
+                "exit_tick_time": 987654321,
+            };
+            expect(Logic.getEndSpotTime(contract_info)).to.eql(987654321);
+        });
+        it('should return contract\'s exit tick time if is_path_dependent is undefined', () => {
+            const contract_info = {
+                "is_path_dependent": false,
+                "sell_spot_time": 123456,
+                "exit_tick_time": 987654321,
+            };
+            expect(Logic.getEndSpotTime(contract_info)).to.eql(987654321);
+        });
+    });
+
+    describe('getFinalPrice', () => {
+        it('should return sell_price as final price when it\'s available', () => {
+            const contract_info = {
+                "sell_price": 12345,
+            };
+            expect(Logic.getFinalPrice(contract_info)).to.eql(12345);
+        });
+        it('should return sell_price as final price when sell_price && bid_price are available', () => {
+            const contract_info = {
+                "sell_price": 12345,
+                "bid_price": 789,
+            };
+            expect(Logic.getFinalPrice(contract_info)).to.eql(12345);
+        });
+        it('should return bid_price as final price when sell_price is not available and bid_price is available', () => {
+            const contract_info = {
+                "bid_price": 789,
+            };
+            expect(Logic.getFinalPrice(contract_info)).to.eql(789);
+        });
+        it('should return 0 as final price when sell_price and bid_price are empty', () => {
+            const contract_info = {
+                "sell_price": false,
+                "bid_price": false,
+            };
+            expect(Logic.getFinalPrice(contract_info)).to.eql(0);
+        });
+    });
+
+    describe('getIndicativePrice', () => {
+        it('should return getFinalPrice if it has final price and contract is ended', () => {
+            const contract_info = {
+                "sell_price": 12345,
+                "status": "closed"
+            };
+            expect(Logic.getIndicativePrice(contract_info)).to.eql(12345);
+        });
+        it('should return null if it doesn\'t final price, bid_price and contract is not ended', () => {
+            const contract_info = {
+                "status": "open",
+            };
+            expect(Logic.getIndicativePrice(contract_info)).to.eql(null);
```

Comment:
can we add another test case where sell_price is empty, bid_price not empty and status is open?